packages/editor/src/components/post-actions/set-as-posts-page.js

```diff
@@ -0,0 +1,158 @@
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { useMemo } from '@wordpress/element';
+import {
+	Button,
+	__experimentalText as Text,
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+} from '@wordpress/components';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
+import { store as noticesStore } from '@wordpress/notices';
+
+/**
+ * Internal dependencies
+ */
+import { getItemTitle } from '../../utils/get-item-title';
+
+const SetAsPostsPageModal = ( { items, closeModal } ) => {
+	const [ item ] = items;
+	const pageTitle = getItemTitle( item );
+	const { currentPostsPage, isPageForPostsSet, isSaving } = useSelect(
+		( select ) => {
+			const { getEntityRecord, isSavingEntityRecord } =
+				select( coreStore );
+			const siteSettings = getEntityRecord( 'root', 'site' );
+			const currentPostsPageItem = getEntityRecord(
+				'postType',
+				'page',
+				siteSettings?.page_for_posts
+			);
+			return {
+				currentPostsPage: currentPostsPageItem,
+				isPageForPostsSet: siteSettings?.page_for_posts !== 0,
+				isSaving: isSavingEntityRecord( 'root', 'site' ),
+			};
+		}
+	);
+
+	const { saveEntityRecord } = useDispatch( coreStore );
+	const { createSuccessNotice, createErrorNotice } =
+		useDispatch( noticesStore );
+
+	async function onSetPageAsPostsPage( event ) {
+		event.preventDefault();
+
+		try {
+			await saveEntityRecord( 'root', 'site', {
+				page_for_posts: item.id,
+				show_on_front: 'page',
+			} );
+
+			createSuccessNotice( __( 'Post page updated' ), {
```

Comment:
```suggestion
			createSuccessNotice( __( 'Post page updated.' ), {
```

Reply:
Done in https://github.com/WordPress/gutenberg/pull/67650/commits/35cea15fecb652b2ba67cd6ecd110a0d7b22ea60. I also made the same change to the homepage action.